funnel/models/event.py

```diff
@@ -48,7 +49,24 @@ class Event(BaseScopedNameMixin, db.Model):
     parent = db.synonym('proposal_space')
     ticket_types = db.relationship('TicketType', secondary=event_ticket_type)
     participants = db.relationship('Participant', secondary='attendee', backref='events', lazy='dynamic')
-    __table_args__ = (db.UniqueConstraint('proposal_space_id', 'name'),)
+    __table_args__ = (db.UniqueConstraint('proposal_space_id', 'name'), db.UniqueConstraint('proposal_space_id', 'title'))
+
+    @classmethod
+    def get(cls, space, name_or_title):
+        return cls.query.filter_by(proposal_space=space)\
+            .filter(or_(cls.name == name_or_title, cls.title == name_or_title)).one_or_none()
```

Comment:
Don't do this. Let the caller be explicit on whether they are querying by name or title. Take `name` and `title` as parameters and do a XOR test on them (via this non-XOR pattern that can extend to multiple inputs, unlike XOR): `if (not not name) + (not not title) != 1: raise TypeError()`

`not not` is a shorthand for `bool()` that avoids the function call.